### DISCUSSION TITLE
How can I permanently add an extra package to the MediaCMS web container on my server?

### DISCUSSION THREAD

**Discussion by Dan1ell** [mediacms-io/mediacms#158] [Q&A] [answered]
I can use docker exec -it <containerid> to install the new package into the running web container, trouble is: I have to repeat this after every execution of docker-compose up. How can I add the extra package permanently?

I see 2 possibilities:

- I can add the package to the docker file, but this does not work because the docker-compose.yaml pulls all the images from repositories, bypassing the docker file. Can I change this?
- I could commit the change into a new image and place that image on DockerHub, then change the yaml file on the server to pull that image instead, but that seems like overkill. 

What is the proper way to add the extra package to the web container?

**Comment by swiftugandan** [comment, marked as answer]
You have to build a new docker image and point docker-compose to that new image ...

**Reply by GeorgiDimitrov87** [reply]
how can I do that? 
Do I have to change image: mediacms/mediacms:latest  in docker-compose.yaml to something else?

**Reply by GeorgiDimitrov87** [reply]
I ended up using docker-compose.dev.yaml configurtion instead then I can add new packages to requirements.txt and requirements.dev.txt
I had to modify the yaml little too
frontend:
    image: node:14
    volumes:
      # - ${PWD}/frontend:/home/mediacms.io/mediacms/frontend/
      - ./frontend:/home/mediacms.io/mediacms/frontend/
    working_dir: /home/mediacms.io/mediacms/frontend/
    command: bash -c "npm install && npm run start"
    env_file:
      # - ${PWD}/frontend/.env
      - ./frontend/.env
    ports:
      - "8088:8088"
    depends_on:
      - web

the selenium_chrome could not register and responds with 404  but the rest  still works
